pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -343,12 +336,48 @@ func (s *NodeController) DoCheck(node *api.Node) []api.NodeCondition {
 			s.deletePods(node.Name)
 		}
 	}
-
 	conditions = append(conditions, *newReadyCondition)
 
+	// Check Condition: NodeSchedulable
+	oldSchedulableCondition := s.getCondition(node, api.NodeSchedulable)
+	newSchedulableCondition := s.checkNodeSchedulable(node)
+	s.updateLastTransitionTime(oldSchedulableCondition, newSchedulableCondition)
+	conditions = append(conditions, *newSchedulableCondition)
+
 	return conditions
 }
 
+// updateLastTransitionTime updates LastTransitionTime for the newCondition based on oldCondition.
+func (s *NodeController) updateLastTransitionTime(oldCondition, newCondition *api.NodeCondition) {
+	if oldCondition != nil && oldCondition.Status == newCondition.Status {
+		// If node status doesn't change, transition time is same as last time.
+		newCondition.LastTransitionTime = oldCondition.LastTransitionTime
+	} else {
+		// Set transition time to Now() if node status changes or `oldCondition` is nil, which
+		// happens only when the node is checked for the first time.
+		newCondition.LastTransitionTime = util.Now()
+	}
+}
+
+// checkNodeSchedulable checks node schedulable condition, without transition timestamp set.
+func (s *NodeController) checkNodeSchedulable(node *api.Node) *api.NodeCondition {
+	if node.Spec.Unschedulable {
+		return &api.NodeCondition{
+			Type:          api.NodeSchedulable,
+			Status:        api.ConditionNone,
+			Reason:        "User specified during node create/update",
```

Comment:
The reason message can't stand on its own, maybe "User marked unschedulable during node create/update"?